services/order.py

```diff
@@ -0,0 +1,41 @@
+from datetime import datetime
+
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import Order, Ticket, MovieSession
+
+
+def create_order(
+        tickets: list[dict],
+        username: str, date:
+        datetime = None
+) -> None:
+    with transaction.atomic():
+        user = get_user_model().objects.get(username=username)
+        order = Order.objects.create(user=user)
+        if date:
+            order.created_at = date
+            order.save()
+
+        ticket_instances = []
+        for ticket_data in tickets:
+            movie_session = MovieSession.objects.get(
+                pk=ticket_data["movie_session"]
+            )
+            ticket_instances.append(Ticket(
+                row=ticket_data["row"],
+                seat=ticket_data["seat"],
+                movie_session=movie_session,
+                order=order
+            ))
+
+        Ticket.objects.bulk_create(ticket_instances)
```

Comment:
This solution look a bit complicated, you definitely benefit from using bulk_create but also you have to create movie_session object on each iteration + bulk_create doesn't trigger full_clean method on create. I'd say more optimal solution here would be something like this:
```
        for ticket_data in tickets:
            1. prepare ticket data (add movie_session_id - just an id no need to create instance, add order)
            2. Ticket.objects.create(**ticket_data)